src/rust/engine/src/externs/interface.rs

```diff
@@ -897,6 +897,20 @@ async fn workunit_to_py_value(workunit: &Workunit, core: &Arc<Core>) -> CPyResul
     externs::store_dict(artifact_entries)?,
   ));
 
+  if let Some(counters) = &workunit.counters {
+    let mut counters_entries = Vec::new();
```

Comment:
We could write this as `let counters_entries: Vec<_> = counters.iter().map(...).collect();`, avoiding a mutable variable.

Reply:
Will do.